test/e2e/tests/tolerate-failure/utils/testSetup.ts

```diff
@@ -0,0 +1,105 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import Timers, { TimerWithId } from './Timers';
+import TimerHelper from './TimersHelper';
+
+export type TimerTestResult = {
+  testName: string;
+  testFile: string;
+  timestamp: string;
+  timers: TimerWithId[];
+};
+
+/**
+ * Performance tracker that collects TimerHelper instances and generates reports
+ */
+class TestPerformanceTracker {
+  private timers: TimerHelper[] = [];
+
+  addTimer(timer: TimerHelper): void {
+    this.timers.push(timer);
+  }
+
+  getAllTimers(): TimerHelper[] {
+    return [...this.timers];
+  }
+
+  clear(): void {
+    this.timers = [];
+  }
+
+  generateReport(testName: string, testFilePath: string): void {
+    try {
+      // Get all timer data from the global Timers singleton
+      const allTimers = Timers.getAllTimers();
+
+      // Only generate report if there are timers
+      if (allTimers.length === 0) {
+        console.log('No timers found for this test, skipping JSON generation');
+        return;
+      }
+
+      // Create test result object using existing TimerWithId type
+      const timerResults: TimerTestResult = {
+        testName,
+        testFile: path.basename(testFilePath),
+        timestamp: new Date().toISOString(),
+        timers: allTimers, // Use the existing TimerWithId[] directly
+      };
+
+      // Create results directory if it doesn't exist
+      const resultsDir = path.join(__dirname, '..', 'results');
```

Comment:
Let's be consistent with other behavior and save somewhere in `test/test-results`

Reply:
Oh sorry, wasn't specific enough.  Now the file is in `/test/e2e/tests/tolerate-failure/test-results`

Other results are in subfolders of `test/test-results`.  Maybe for this `test/test-results/power-user-scenarios` would make sense?